app/src/androidTestWebview/java/org/mozilla/focus/activity/SearchEngineSelectionTest.java

```diff
@@ -60,85 +74,140 @@ public void tearDown() throws Exception {
-        TestHelper.settingsHeading.waitForExists(waitingTime);
-        TestHelper.pressBackKey();
-
-        /* load blank spaces and press enter key for search, it should not do anything */
-        TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime);
-        TestHelper.inlineAutocompleteEditText.clearTextField();
-        TestHelper.inlineAutocompleteEditText.setText("   ");
-        TestHelper.pressEnterKey();
-        assertTrue(TestHelper.inlineAutocompleteEditText.exists());
-
-        /* Now do some search */
-        TestHelper.inlineAutocompleteEditText.clearTextField();
-        TestHelper.inlineAutocompleteEditText.setText("mozilla focus");
-        TestHelper.hint.waitForExists(waitingTime);
-
-        // Check the search hint bar is correctly displayed
-        assertTrue(TestHelper.hint.getText().equals("Search for mozilla focus"));
-        TestHelper.hint.click();
-
-        /* Browser shows google search webview*/
-        googleWebView.waitForExists(waitingTime);
-        TestHelper.progressBar.waitForExists(webPageLoadwaitingTime);
-        Assert.assertTrue(TestHelper.progressBar.waitUntilGone(webPageLoadwaitingTime));
-        assertTrue (TestHelper.browserURLbar.getText().contains("google"));
-        assertTrue (TestHelper.browserURLbar.getText().contains("mozilla"));
-        assertTrue (TestHelper.browserURLbar.getText().contains("focus"));
-
-        /* tap url bar, and check it displays search term instead of URL */
-        TestHelper.browserURLbar.click();
-        TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime);
-        assertEquals(TestHelper.inlineAutocompleteEditText.getText(), "mozilla focus");
-        TestHelper.pressEnterKey();
-        googleWebView.waitForExists(waitingTime);
-        TestHelper.progressBar.waitForExists(webPageLoadwaitingTime);
-        Assert.assertTrue(TestHelper.progressBar.waitUntilGone(webPageLoadwaitingTime));
-        assertTrue (TestHelper.browserURLbar.getText().contains("google"));
-        assertTrue (TestHelper.browserURLbar.getText().contains("mozilla"));
-        assertTrue (TestHelper.browserURLbar.getText().contains("focus"));
-
-         /* Now do another search */
-        TestHelper.browserURLbar.click();
-        TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime);
-        TestHelper.inlineAutocompleteEditText.clearTextField();
-        TestHelper.inlineAutocompleteEditText.setText("mozilla focus");
-        TestHelper.hint.waitForExists(waitingTime);
-
-        // Check the search hint bar is correctly displayed
-        assertTrue(TestHelper.hint.getText().equals("Search for mozilla focus"));
-        TestHelper.hint.click();
+
+        // Open [settings menu] and select Search
+        {
+            assertTrue(TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime));
+            openSettings();
+            searchEngineSelectorLabel.click();
+        }
```

Comment:
for ease of debug, it always helps to check whether the UI action caused intended result.  so in this case, you could check whether the screen really got updated after .click().  this applies to all the other ui actions below

Reply:
so, do a waitForExists() and then an assert to verify?